### PR TITLE
3.x: use union types

### PR DESCRIPTION
**Pull request by Indianer3c**

### CODE REVIEW

File: docs/en/request-authorization-middleware.rst [hunk 1]
```diff
@@ -33,7 +33,7 @@ and add::
          * @param \Cake\Http\ServerRequest $request Server Request
          * @return bool
          */
-        public function canAccess($identity, ServerRequest $request)
+        public function canAccess($identity, ServerRequest $request): ResultInterface
```
Comment (dereuromark) [inline docs/en/request-authorization-middleware.rst]: Why does a bool method (and documented as such) now return an object?

Comment (LordSimal) [reply docs/en/request-authorization-middleware.rst]: this doc just needs to be updated to represent https://github.com/cakephp/authorization/blob/3.x/src/Policy/RequestPolicyInterface.php#L34

Comment (Indianer3c) [reply docs/en/request-authorization-middleware.rst]: Sorry my mistake

**Comment by LordSimal** [comment]
thank you 👍🏻

**Pull request merged**